service/block/store.go

```diff
@@ -0,0 +1,21 @@
+package block
+
+import (
+	"context"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-core/pkg/da"
+	"github.com/celestiaorg/celestia-node/ipld"
+)
+
+func (s *Service) StoreBlockData(ctx context.Context, data *ExtendedBlockData) error {
+	shares := ipld.ConvertEDStoShares(data)
+	// TODO @renaynay: it's inefficient that we generate the EDS twice
```

Comment:
Where is the 2nd place the EDS gets generated? In core? That is OK but it should not be necessary to generate it in node twice IMO.

Reply:
Can you comment that it's the IPLD that extends the shares again?

Reply:
> we have to convert the extended block back to shares via ConvertEDStoShares and then PutData extends those shares again...... :(

let me check this out locally before we merge this. I thought ConvertEDStoShares is just a different representation of the already extended (erasured) block. And PutData just takes these shares (and does not do any erasure coding again).

Reply:
Ah, NVM I see this erasure codes the block again:
https://github.com/celestiaorg/celestia-node/blob/b0e075914b0d76ecccbd24d5cd6efe3b9431ab9b/ipld/write.go#L17

Let's just open an issue about fixing that.

Maybe rename `ConvertEDStoShares` to something clearer. maybe `ExtractODSShares`.